using/self_service.md

```diff
@@ -322,3 +322,118 @@ Once allocation has been created, you can see its access information:
+
+Before using JupyterHub, please setup [Python management service](#python-management-service) on a VM.
+
+Once provisioned, JupyterHub configuration file is located in ```/etc/jupyterhub/jupyterhub_config.py```. All changes manually applied to JupyterHub configuraiton files will not be reflected in JupyterHub details form. Moreover, each time you click "Apply configuration" button, the existing configuration file is completely replaced with the new one generated out of the currently saved state in ETAIS. However, the backup for old configuration file is created in the same folder.
+
+JupyterHub management service is not notified if you change, for example, admin rights through JupyterHub GUI - do it through self-service.
+
+### Details screen - manual JupyterHub user entry
+
+![JupyterHub management details Linux PAM authentication section](../images/jupyterhub-management-linux-pam.png)
+
+1. Virtual environments come from Python management service.
+2. It is possible to make these virtual environments accessible as Jupyter kernels. All JupyterHub users will have access to them. However, they will not be allowed to install or remove libraries through notebook or SSH in these kernels.
+3. There are 2 types of authentication methods available. In both cases JupyterHub users correspond to Linux system users.
+ * Linux PAM: you need to manually define JupyterHub users. Corresponding system users will be created once Apply configuration button is pressed.
+ * OAuth: see the next section. Corresponding system users will be created on first login.
+4. Apply configuration (currently active Jupyter users will experience short connection outage with JupyterHub).
+5. Passwords for already created users are not stored in ETAIS database. Password should be defined only for new users (form validation will guide you).
+
+### Details screen - OAuth2 authentication method
+
+![JupyterHub management details OAuth2 authentication section](../images/jupyterhub-management-oauth.png)
+
+1. Currently there are 2 supported OAuth providers: GitLab and Microsoft Azure.
+2. External link to a GitHub page where it is explained how to configure authentication for each OAuth provider.
+3. At least one admin should be specified. All special characters in usernames will be replaced with underscores _ once config is saved.
+4. By default, any OAuth user can login to JupyterHub. If you want to prevent that from happening, you can specify that only specified admins are eligible to log in.
+5. If you want to add additional users who are allowed to log in, specify them here.
+
+### How to restart JupyterHub via SSH
+
+If you need to manually restart JupyterHub SSH, execute the following commands. Normally, there is no need in doing that, since when JupyterHub details screen is saved, all changes will be applied and JupyterHub restarted).
+
+Stop JupyterHub process:
+
+```supervisorctl stop jupyterhub```
+
+Kill configurable-http-proxy process:
+
+```sudo pkill node```
```

Comment:
Why not use systemd? Why do we need sudo here and don't need sudo in two other commands?

Reply:
jupyterhub doesn't kill its proxy properly

Reply:
Ok, please explanation to docs, it would be very useful addition.